### PR TITLE
Implement the 'Borrow Series' modal

### PR DESCRIPTION
**Pull request by bp323**
Implement the 'Borrow Series' modal per design. This depends on #112.

screen shot 2015-01-16 at 15 04 32
screen shot 2015-01-16 at 15 04 48
screen shot 2015-01-16 at 15 05 03

### CODE REVIEW

**Comment by bp323** [comment]
Submitting a PR for this but going to go through a round of self-review first.

**Comment by bp323** [comment]
Finishes self-review, over to @Coenego

File: apps/timetable/admin/ui/css/skin.css [hunk 1]
```diff
@@ -142,3 +142,17 @@
...
+    background-color: #FFF;
+}
+
+#gh-borrow-series-modal .list-group .list-group-item .gh-borrow-series-borrowed .gh-list-description p {
+    color: #171717;
+}
+
+#gh-borrow-series-modal .list-group .list-group-item i.fa-check {
+    color: #42305B;
+}
```
Comment (Coenego) [inline apps/timetable/admin/ui/css/skin.css]: Empty line required to be consistent with the rest of the files

File: shared/gh/js/gh.borrow-series.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+     *
+     * @private
+     */
+    var showBorrowSeriesModal = function() {
+        // Get the ID of the module to borrow series to
+        var moduleId = $(this).closest('.list-group-item').data('id');
+        // Get the part ID of the module
+        var partId = $(this).closest('#gh-modules-list-container').data('partid');
+
+
```
Comment (Coenego) [inline shared/gh/js/gh.borrow-series.js]: This empty line can probably go

File: shared/gh/js/gh.borrow-series.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            // Show the modal
+            $('#gh-borrow-series-modal').modal();
+        });
+    };
+
+    /**
+     * Add a series to the list of series that will be borrowed by applying the `gh-borrow-series-borrowed`
+     * class to its container
+     */
```
Comment (Coenego) [inline shared/gh/js/gh.borrow-series.js]: `@private` needed

File: shared/gh/js/gh.borrow-series.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    var addSeriesToBorrow = function() {
+        $(this).closest('.list-group-item ').addClass('gh-borrow-series-borrowed');
+    };
+
+    /**
+     * Remove a series from the list of series that will be borrowed by removing the
+     * `gh-borrow-series-borrowed` class from its container
+     */
```
Comment (Coenego) [inline shared/gh/js/gh.borrow-series.js]: `@private` needed

File: shared/gh/js/gh.borrow-series.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    var removeSeriesToBorrow = function() {
+        $(this).closest('.list-group-item ').removeClass('gh-borrow-series-borrowed');
+    };
+
+    /**
+     * Collect all series marked as 'to borrow' and borrow them into a module
+     */
```
Comment (Coenego) [inline shared/gh/js/gh.borrow-series.js]: `@private` needed

File: shared/gh/api/gh.api.util.js [hunk 6]
```diff
@@ -484,14 +486,21 @@ define(['exports', 'moment', 'sinon', 'bootstrap-notify'], function(exports, mom
             'declarePartial': function(name, template) {
                 partialCache[name] = _.template(template);
             },
-            'partial': function(name, data) {
+            'partial': function(name, data, renderAtStart) {
+                // TODO: replace `renderStart` with a more robust solution for delayed rendering
+                //       of partials inside of partials
+                if (renderAtStart === false) {
+                    return '<%= _.partial("' + name + '", {data: data}, null) %>';
+                }
```
Comment (Coenego) [inline shared/gh/api/gh.api.util.js]: It looks like the code coverage decreased, caused by the above-stated changes.

Comment (bp323) [inline shared/gh/api/gh.api.util.js]: I'll just ignore the if for now, we should come up with a better approach to this.

**Comment by Coenego** [comment]
Reassigning to @bp323 for follow-up

**Comment by bp323** [comment]
Back to @Coenego for review.

**Pull request merged**